web/beacon-app/src/features/projects/schemas/createProjectTopicSchema.ts

```diff
@@ -0,0 +1,26 @@
+import { t } from '@lingui/macro';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+
+import { NewTopicDTO } from '../types/createTopicService';
+
+export const createProjectTopicSchema = Yup.object().shape({
+  topic_name: Yup.string()
+    .trim()
+    .required(t`Topic name is required.`)
+    .matches(/^[^\s]*$/, t`Topic name cannot include spaces.`)
```

Comment:
we could remove the space directly by reformatting the value with yup
```suggestion
    .transform((value, originalValue) => {
        if(!originalValue.test(regex)) {
          return originalValue.trim()
          }
          return value
        }
```

Reply:
Yes, we could remove the space, but this may confuse users if they enter a name that includes spaces and then see it saved without the spaces. The modal does state that spaces cannot be included, but some users may not read all of the text. Showing a validation error message, if the topic name does include a space, should help prevent confusion.